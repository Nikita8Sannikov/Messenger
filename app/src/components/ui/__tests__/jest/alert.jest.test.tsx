import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import AlertComponent from "@components/ui/alert";

describe("AlertComponent", () => {
    test("should render Alert", () => {
        render(<AlertComponent show={true} status="success" onClose={() => { }} severity="success">Test Alert</AlertComponent>);
        const alertElement = screen.getByText("Test Alert");
        expect(alertElement).toBeInTheDocument();
    }
    );

    test("uses default status when not provided", () => {
        render(<AlertComponent show={true}>Alert Content</AlertComponent>);
        const alert = screen.getByRole("alert");
        expect(alert).toHaveClass("MuiAlert-colorSuccess");
    });

    test("should call onClose when close button is clicked", async () => {
        const user = userEvent.setup();
        const handleClose = jest.fn();
        render(<AlertComponent show={true} status="success" onClose={handleClose} severity="success">Test Alert</AlertComponent>);
        const closeButton = screen.getByRole("button");
        await user.click(closeButton);
        expect(handleClose).toHaveBeenCalledTimes(1);
    }
    );

    test("matches snapshot", () => {
        const { asFragment } = render(<AlertComponent></AlertComponent>);

        // Снимаем снапшот компонента
        expect(asFragment()).toMatchSnapshot();
    });
});