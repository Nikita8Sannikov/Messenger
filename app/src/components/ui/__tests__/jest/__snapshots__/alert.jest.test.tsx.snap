// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertComponent matches snapshot 1`] = `<DocumentFragment />`;
